core/src/main/java/org/elasticsearch/index/reindex/BulkByScrollTask.java

```diff
@@ -43,32 +43,127 @@
+        }
+
+        if (isSliceChild()) {
+            return childWorker.getStatus();
+        }
+
+        return emptyStatus();
+    }
+
+    private BulkByScrollTask.Status emptyStatus() {
+        return new Status(Collections.emptyList(), getReasonCancelled());
+    }
+
     /**
-     * The number of sub-slices that are still running. {@link WorkingBulkByScrollTask} will always have 0 and
-     * {@link ParentBulkByScrollTask} will return the number of waiting tasks. Used to decide how to perform rethrottling.
+     * Returns true if this task is a parent of other sliced tasks. False otherwise
      */
-    public abstract int runningSliceSubTasks();
+    public boolean isSlicesParent() {
+        return parentWorker != null;
+    }
 
     /**
-     * Apply the {@code newRequestsPerSecond}.
+     * Sets this task to be a parent task for {@code slices} sliced subtasks
      */
-    public abstract void rethrottle(float newRequestsPerSecond);
+    public void setSliceChildren(int slices) {
+        if (isSlicesParent()) {
+            throw new IllegalStateException("Parent worker is already set");
+        }
+        if (isSliceChild()) {
+            throw new IllegalStateException("Worker is already set as child");
+        }
 
-    /*
-     * Overridden to force children to return compatible status.
+        parentWorker = new ParentBulkByScrollWorker(this, slices);
+    }
+
+    /**
+     * Returns the worker object that tracks the state of sliced subtasks. Throws IllegalStateException if this task is not set to be
+     * a parent task.
      */
-    public abstract BulkByScrollTask.Status getStatus();
+    public ParentBulkByScrollWorker getSlicesParentWorker() {
+        if (!isSlicesParent()) {
+            throw new IllegalStateException("This task is not set as a parent worker");
+        }
+        return parentWorker;
+    }
 
     /**
-     * Build the status for this task given a snapshot of the information of running slices.
+     * Returns true if this task is a child task that performs search requests. False otherwise
      */
-    public abstract TaskInfo getInfoGivenSliceInfo(String localNodeId, List<TaskInfo> sliceInfo);
+    public boolean isSliceChild() {
+        return childWorker != null;
+    }
+
+    /**
+     * Sets this task to be a child task that performs search requests, when the request is not sliced.
+     * @param requestsPerSecond How many search requests per second this task should make
+     */
+    public void setSliceChild(float requestsPerSecond) {
+        setSliceChild(requestsPerSecond, null);
+    }
+
+    /**
+     * Sets this task to be a slice child task that performs search requests
+     * @param requestsPerSecond How many search requests per second this task should make
+     * @param sliceId If this is is a sliced task, which slice number this task corresponds to
+     */
+    public void setSliceChild(float requestsPerSecond, Integer sliceId) {
```

Comment:
There are a couple of `@Nullable`s on the classpath. The one in `org.elasticsearch.common` is probably the right one.